.github/workflows/ci.yml

```diff
@@ -313,12 +319,14 @@ jobs:
           CARGO: cargo-zigbuild
           CARGO_BUILD_TARGET: aarch64-unknown-linux-gnu
           PYO3_CROSS_LIB_DIR: /opt/python/cp38-cp38/lib
+          DIST_EXTRA_CONFIG: /tmp/build-opts.cfg
         run: |
           mkdir -p $PYO3_CROSS_LIB_DIR
           docker cp -L $(docker create --rm quay.io/pypa/manylinux2014_aarch64:latest):/opt/python/cp38-cp38 /opt/python
           cd examples/namespace_package
-          python -m pip install wheel
-          python setup.py bdist_wheel --plat-name manylinux2014_aarch64
+          python -m pip install build wheel
+          echo -e "[bdist_wheel]\nplat_name=manylinux2014_aarch64" > $DIST_EXTRA_CONFIG
+          python -m build --no-isolation
```

Comment:
Potentially you could also rework this test to use the `rust_with_cffi` example and keep using `setup.py` for now?

Reply:
@davidhewitt Since `distutils` impose as restriction that users cannot set `plat_name` directly on `build`, is it OK for `setuptools-rust` to be deriving its own value of `plat_name` from `build` when comparing the host and target architecture? Or I am misinterpreting the code?

Reply:
It seems that changing the following [fixes the problem with `test-zigbuild`](https://github.com/PyO3/setuptools-rust/actions/runs/5857228514/job/15878634313?pr=352):

```diff
--- a/setuptools_rust/setuptools_ext.py
+++ b/setuptools_rust/setuptools_ext.py
@@ -163,9 +163,11 @@ def add_rust_extension(dist: Distribution) -> None:
-                options = self.distribution.get_cmdline_options().get("bdist_wheel", {})
-                plat_name = options.get("plat-name") or self.plat_name
+                bdist_wheel = self.distribution.get_command_obj("bdist_wheel")
+                plat_name = bdist_wheel.plat_name or self.plat_name
```

(probably related to the fact that when the test calls `python -m build`, the `plat_name` is given via a `DIST_EXTRA_CONFIG` file instead of `sys.argv`, but I might be wrong).